flows/actions/classify_nlu.go

```diff
@@ -0,0 +1,130 @@
+
+// ClassifyNLUAction can be used to classify the intent and entities from a given input using an NLU classifier. It always
+// saves a result indicating whether the classification was successful, skipped or failed, and what the extracted intents
+// and entities were.
+//
+//   {
+//     "uuid": "8eebd020-1af5-431c-b943-aa670fc74da9",
+//     "type": "classify_nlu",
+//     "classifier": {
+//       "uuid": "1c06c884-39dd-4ce4-ad9f-9a01cbe6c000",
+//       "name": "Booking"
+//     },
+//     "input": "@input.text",
+//     "result_name": "Intent"
+//   }
+//
+// @action classify_nlu
+type ClassifyNLUAction struct {
+	baseAction
+	onlineAction
+
+	Classifier *assets.ClassifierReference `json:"classifier" validate:"required"`
+	Input      string                      `json:"input" validate:"required" engine:"evaluated"`
+	ResultName string                      `json:"result_name" validate:"required"`
+}
+
+// NewClassifyNLU creates a new classify NLU action
+func NewClassifyNLU(uuid flows.ActionUUID, classifier *assets.ClassifierReference, input string, resultName string) *ClassifyNLUAction {
+	return &ClassifyNLUAction{
+		baseAction: newBaseAction(TypeClassifyNLU, uuid),
+		Classifier: classifier,
+		Input:      input,
+		ResultName: resultName,
+	}
+}
+
+// Execute runs this action
+func (a *ClassifyNLUAction) Execute(run flows.FlowRun, step flows.Step, logModifier flows.ModifierCallback, logEvent flows.EventCallback) error {
+	classifiers := run.Session().Assets().Classifiers()
+	classifier := classifiers.Get(a.Classifier.UUID)
+
+	// substitute any variables in our input
+	input, err := run.EvaluateTemplate(a.Input)
+	if err != nil {
+		logEvent(events.NewError(err))
+	}
+
+	classification, skipped, err := a.classify(run, step, input, classifier, logEvent)
+	if err != nil {
+		logEvent(events.NewError(err))
+
+		if skipped {
+			a.saveSkipped(run, step, input, logEvent)
+		} else {
+			a.saveFailure(run, step, input, logEvent)
+		}
+	} else {
+		a.saveSuccess(run, step, input, classification, logEvent)
+	}
+
+	return nil
+}
+
+func (a *ClassifyNLUAction) classify(run flows.FlowRun, step flows.Step, input string, classifier *flows.Classifier, logEvent flows.EventCallback) (*flows.NLUClassification, bool, error) {
+	if input == "" {
+		return nil, true, errors.New("can't classify empty input, skipping classification")
+	}
+	if classifier == nil {
+		return nil, false, errors.Errorf("missing %s", a.Classifier.String())
+	}
+
+	nluSvc := run.Session().Engine().Services().NLU(run.Session(), classifier)
+	if nluSvc == nil {
+		return nil, false, errors.Errorf("no NLU provider available")
+	}
+
+	classification, err := nluSvc.Classify(run.Session(), input)
+	return classification, false, err
+}
+
+func (a *ClassifyNLUAction) saveSuccess(run flows.FlowRun, step flows.Step, input string, classification *flows.NLUClassification, logEvent flows.EventCallback) {
+	// value is name of top ranked intent if there is one
```

Comment:
this differs from what I had in the examples in the issue but feels more natural (also not actually used by the router tests)

Reply:
makes sense